index.js

```diff
@@ -105,20 +105,52 @@ class IOSDeviceLib extends EventEmitter {
 				return reject(new Error("No arguments provided"));
 			}
 
+			let timer = null;
+			let eventHandler = null;
+			let deviceLostHandler = null;
 			const id = uuid.v4();
-			const eventHandler = (message) => {
+			const removeListeners = () => {
+				if (eventHandler) {
+					this._iosDeviceLibStdioHandler.removeListener(Constants.DataEventName, eventHandler);
+				}
+
+				if (deviceLostHandler) {
+					this._iosDeviceLibStdioHandler.removeListener(Constants.DeviceLostEventName, deviceLostHandler);
+				}
+			};
+
+			const handleMessage = (message) => {
+				removeListeners();
+				message.error ? reject(message.error) : resolve(message);
+			};
+
+			deviceLostHandler = (device) => handleMessage({ error: new Error(`Device ${device.deviceId} lost during operation ${methodName} for message ${id}`) });
+
+			eventHandler = (message) => {
 				if (message && message.id === id) {
+					if (timer) {
+						clearTimeout(timer);
+					}
+
 					delete message.id;
 					if (options && options.shouldEmit) {
 						this.emit(Events.deviceLogData, message);
 					} else {
-						message.error ? reject(message.error) : resolve(message);
-						this._iosDeviceLibStdioHandler.removeListener(Constants.DataEventName, eventHandler);
+						handleMessage(message);
 					}
 				}
 			};
 
+			if (this._options.timeout && !options.disregardTimeout) {
```

Comment:
The `!options.disregardTimeout` check will throw an error if `this._options.timeout` is set because the `options` object will be undefined for all methods except for `startDeviceLog`.

Reply:
Nice catch, I've added a default value for options